### ISSUE TITLE
Progression system could use some polishing, and a few ideas on how to do that

### ISSUE THREAD

**Issue by Jack-0-Spades** [Electroblob77/Wizardry#480]
I really like this mod and all of it's features but there are several things about the progression of the mod that could use some polish. I have a few ideas that I hope will be useful. To summarize it in advanced I think there needs to be more options in the various progression systems and more rewards for investing in them. Too narrow of a progression will make players feel the grind all the more, while giving them more options and rewards will help them feel the grind less because they can swap to another option if one starts getting particularly annoying.

Spell progression: I read a comment in another post that there is supposed to be a higher chance of finding a spell if you don't already have it, however I think there should be more options for getting spells. For example the player could find research fragments and research journals that could be used to get new spells that they don't already know. These research journals left behind by other wizards could be used for other means as well like finding artifacts or leveling wands I will explain more below.

Wand progression: I am unable to play with the current wand progression system and almost always turn on the legacy option after a bit. The grind just does not feel satisfying, instead I feel as though the fun is sucked out of the mod when playing with it on. Honestly, I think it needs to be completely re-imagined. I think at least part of the requirements for upgrading a wand should be linked to player progression. You are playing as a wizard so it doesn't feel right if you do not have any magic abilities inherent to yourself. The previously mentioned research journals could also be used for various means like gaining magic XP (learning from the experiences of other wizards) or even upgrading your wand. 

Artifact progression: Rather than adding a new way to get artifacts I think having a way to find shrines would be better. A treasure hunting system could be created using a simple puzzle minigame or something to find the closest undiscovered shrine. I read in another post you were also thinking about making it so that you would  be more likely to find new artifacts that you did not already have. I think the two ideas would compliment each other. Research journals could also be needed for this as this would force the player to make a choice about what they want to use it for.

**Comment by Electroblob77** [comment]
Thanks! I am aware that the wand progression system needs some work. The original intent was to take some of the randomness out of how much play time it takes to get to a given tier, because RNG can also be a grind, and at least now you know how much grinding is needed. But yes, ideally it shouldn't feel grind-y at all.

More options for getting spells: I'm adding some more structures to the upcoming 4.3 update, which will help simply because there will be more structures overall, and I'm also planning to add spell books to fishing loot. I did consider some sort of spell page/fragment system (in quite some depth, actually) but ultimately decided it would overcomplicate things, as I know a lot of people like the mod for its simplicity.

Progression: First time round, it doesn't _actually_ take that long. I've playtested it and adjusted it down several times. However, once you get to a second wand it does take a while, so the new update will add a substantial progression bonus for tiers you've already reached, which is sort of like 'experience' within the player themselves.

Finding stuff: Part of the point of having Antique Atlas integration was exactly this. However, I'd also like to add vanilla-style explorer maps or something similar to help with this. The rarity of everything might be reduced a bit by default as well.
